mllib-local/pom.xml

```diff
@@ -61,13 +61,17 @@
       This spark-tags test-dep is needed even though it isn't used in this module, otherwise testing-cmds that exclude
       them will yield errors.
     -->
+    <dependency>
```

Comment:
This has to be removed; it can't depend on core.
I think that means you need to move the property out of core.

Reply:
Without dependency of core, ConfigBuilder/ConfigEntry won't be accessible. So I'd propose to add an environment variable in spark-env.sh. see new commit for details.

Reply:
Ah, right. OK, in that case I would leave the property in `core`, but reference it only by name (unfortunately) here in `mllib-local`, with a comment about why. You can remove `.ml` from the name too; it's not specific to ML really.

I don't think env variables are a good solution here.

Reply:
hmm, instead env variables, we can pass this configuration via blas property file considering ```mllib-local``` package does not have access to ```core``` module.

Reply:
A property file is even more convoluted. You can still access the property; you just can't use the config API or symbol.

Reply:
If we leave the property in core, we need to import org.apache.spark.internal.config to reference it in mllib-local thus still can't avoid dependency of core module.  Another option is that we implement a logic to directly read this value from spark-defaults.conf through method like System.getProperties, which will need no modification and dependency of core module.

Reply:
You don't, you just need to reference the config key by name, without references to symbols in core. This is a fine compromise.

Reply:
Do you mean get the value using SparkConf by key? But SparkConf also has dependency on core module.

Reply:
Ah, OK that is not even available, right. I think I'd just make it a hard-coded constant then. I don't think we can reasonably expect people to find and discover and env variable that probably isn't that meaningful to tune. Just set it conservatively. 512?

Reply:
Hi @srowen, how about 256 which we have finalized in previous discussions?

Reply:
Sure, I'm just trying to think of a threshold that should easily be a win regardless of other factors.

Reply:
revert configurable related commits and now use 256 as the nativeL1Threshold.